src/nu/validator/htmlparser/impl/Tokenizer.java

```diff
@@ -6189,6 +6195,19 @@ private void ensureBufferSpace(int inputLength) throws SAXException {
                         case '>':
                             state = transition(state, Tokenizer.DATA,
                                     reconsume, pos);
+                            // Processing instruction syntax goes through these
+                            // states only in Gecko's XML View Source--not in HTML
+                            // parsing in Java or in Gecko.
+                            // Since XML View Source doesn't use the
+                            // suspension-after-current-token facility, its extension
+                            // to processing instruction states is strictly unnecessary
+                            // at the moment. However, if these states ever were to be
+                            // used together with the suspension-after-current-token
+                            // facility, these states would need to participate, since
+                            // suspension could be requested when only less-than has been
+                            // seen and we don't yet know if we end up here. Handling
+                            // the currently unnecessary case in order to avoid leaving
+                            // a trap for future modification.
```

Comment:
```suggestion
                            // Processing instruction syntax goes through these
                            // states only in Gecko's XML View Source--not in HTML
                            // parsing in Java or in Gecko.
                            // Since XML View Source doesn't use the
                            // suspension-after-current-token facility, its extension
                            // to processing-instruction states is strictly unnecessary
                            // at the moment. However, if these states ever were to be
                            // used together with the suspension-after-current-token
                            // facility, these states would need to participate, since
                            // suspension could be requested when only less-than has been
                            // seen and we don't yet know if we end up here. Handling
                            // the currently-unnecessary case in order to avoid leaving
                            // a trap for future modification.
```